modules/weboramaRtdProvider.js

```diff
@@ -716,94 +707,24 @@ class WeboramaRtdProvider {
-   * @param {Object} bid.params
-   * @param {Object} bid.params.keyword
-   * @param {Profile} profile
-   * @returns {void}
-   */
-  // eslint-disable-next-line no-dupe-class-members
-  #handleAppnexusBid(bid, profile) {
-    const base = 'params.keywords';
-    this.#assignProfileToObject(bid, base, profile);
-  }
-
-  /** handle pubmatic bid
-   * @method
-   * @private
-   * @param {Object} bid
-   * @param {Object} bid.params
-   * @param {string} bid.params.dctr
-   * @param {Profile} profile
-   * @returns {void}
-   */
-  // eslint-disable-next-line no-dupe-class-members
-  #handlePubmaticBid(bid, profile) {
-    const sep = '|';
-    const subsep = ',';
-
-    bid.params ||= {};
-
-    const data = bid.params.dctr || '';
-    const target = new Set(data.split(sep).filter((x) => x.length > 0));
-
-    Object.entries(profile).forEach(([key, values]) => {
-      const value = values.join(subsep);
-      const keyword = `${key}=${value}`;
-      target.add(keyword);
-    });
-
-    bid.params.dctr = Array.from(target).join(sep);
-  }
-
-  /** handle smartadserver bid
-   * @method
-   * @private
-   * @param {Object} bid
-   * @param {Object} bid.params
-   * @param {string} bid.params.target
-   * @param {Profile} profile
-   * @returns {void}
-   */
-  // eslint-disable-next-line no-dupe-class-members
-  #handleSmartadserverBid(bid, profile) {
-    const sep = ';';
-
-    bid.params ||= {};
-
-    const data = bid.params.target || '';
-    const target = new Set(data.split(sep).filter((x) => x.length > 0));
-
-    Object.entries(profile).forEach(([key, values]) => {
-      values.forEach(value => {
-        const keyword = `${key}=${value}`;
-        target.add(keyword);
-      })
-    });
-
-    bid.params.target = Array.from(target).join(sep);
-  }
-
-  /** handle rubicon bid
+  /** handle site legacy keywords bidders like appnexus/xandr
    * @method
    * @private
+   * @param {Object} reqBidsConfigObj
+   * @param {Object} reqBidsConfigObj.ortb2Fragments
+   * @param {Object} reqBidsConfigObj.ortb2Fragments.bidder
    * @param {Object} bid
-   * @param {string} bid.bidder
+   * @param {Object} bid.parameters
    * @param {Profile} profile
    * @param {dataCallbackMetadata} metadata
    * @returns {void}
    */
   // eslint-disable-next-line no-dupe-class-members
-  #handleRubiconBid(bid, profile, metadata) {
-    if (isBoolean(metadata.user)) {
-      const section = metadata.user ? 'visitor' : 'inventory';
-      const base = `params.${section}`;
-      this.#assignProfileToObject(bid, base, profile);
+  #handleSiteLegacyKeywordsBidders(reqBidsConfigObj, bid, profile, metadata) {
+    if (metadata.user) {
+      this.#setBidderOrtb2(reqBidsConfigObj.ortb2Fragments?.bidder, bid.bidder, 'user.keywords', profile);
     } else {
-      logMessage(`SKIP bidder '${bid.bidder}', data from '${metadata.source}' is not defined as user or site-centric`);
+      this.#assignProfileToObject(bid, 'params.keywords', profile);
```

Comment:
I'm trying to follow the updates that were made in this area with the recent commits.  

If this code is still primarily there for the appnexus bid adapter (as implied by the `LEGACY_SITE_KEYWORDS_BIDDERS` if check made earlier to call this `handleSiteLegacyKeywordsBidders` function), you could move it or generalize it further by adding this data to the equivalent ortb2 fields (like site.keywords) instead of the bidder.params location.  Our adapter should be able to read that data from the ortb2 location, and it should allow other bidders to read that data as well.

If this is serving some other purpose still, can you please clarify?  Thanks.

Reply:
Hello

I just copy this `LEGACY_SITE_KEYWORDS_BIDDERS` from `1plusXRtdProvider.js`

Reply:
I'm not sure I follow - the 1plusXRtdProvider appears to be writing a modified form of segment data into the appnexus auction keywords field (which they technically don't need to do anymore, since they could write it to the ortb2.site.keywords field instead).  

So while I may understand if you want to do an extra step/conversion of your data to a keyword-like format (like 1plusXRtdProvider), you shouldn't need to target the bidder keywords field like they do.  

The only reason I'm bringing up this point was that the PR seemed to be designed to move your data to the ortb2 fields.  So that should be possible even for us (appnexus), instead of using a bidder param field (which feels like it's against the grain).

Reply:
so if I understand well, in the case of appnexis I must push to `ortb2.site.keywords` or `ortb2.user.keywords` depending if it is site-centric or user-centric, right?

Reply:
For our endpoint, we just have either request level keywords or adunit/placement specific keywords.  

For our request level keywords they read the ortb2 keyword fields from various sources, collating them together based on keynames/keyvalues to send along in the auction request.  For the adunit keywords, we read from the adUnit ortb2Imp field that publishers can choose to populate, if they have unique data for that slot.  So it doesn't as much matter which of the fields are used in the ortb2 keywords properties, we'll check them and pull in the data if it's there.

Reply:
how about now: db1fc089081660753a0a15348508c6be7aba42ff ?

Reply:
Does it make sense to use this library and base on prebid 8 ? [libraries/appnexusKeywords/anKeywords.js](https://github.com/prebid/Prebid.js/pull/10071/files#diff-ac43bc058ac47dc8abc3602fe21d9b07ebb2adde809fac523d55099618c3166d)

Reply:
I did some digging on this and the problems are related, `keywords` as set from this revision are removed by the fpdValidation module because of incorrect type:

![image](https://github.com/prebid/Prebid.js/assets/10151381/f923a3e4-8da4-4636-b080-ca3c3484b9c6)